packages/web-pkg/src/components/ItemFilter.vue

```diff
@@ -204,18 +215,26 @@ export default defineComponent({
     })
 
     return {
-      queryParam,
-      filterInputRef,
-      itemFilterListRef,
-      selectedItems,
       clearFilter,
-      toggleItemSelection,
-      isItemSelected,
       displayedItems,
+      filterInputRef,
       filterTerm,
+      isItemSelected,
       isSelectionAllowed,
+      itemFilterListRef,
+      queryParam,
+      selectedItems,
       setDisplayedItems,
-      showDrop
+      showDrop,
+      toggleItemSelection
+    }
+  },
+  computed: {
+    getPlaceholder() {
+      return this.placeholder === '' ? this.$gettext('Enter term') : this.placeholder
+    },
+    getLabel() {
+      return this.label === '' ? this.$gettext('Filter list') : this.label
```

Comment:
> Declaring computed inside the setup() method threw an error with the $gettext method (not defined).

You need to make use of the Gettext-composable first within `setup`: `const { $gettext } = useGettext()`.

> Also, I always try to separate layout definitions (inside template tags) and functional definitions (inside script tag).

While I agree for most situations, such simple and label-related checks are fine in the template IMO. Plus you save yourself from finding a name for the computed props, because the `get...` prefix like you used before is usually reserved for methods.